tinygrad/renderer/cstyle.py

```diff
@@ -72,7 +72,7 @@ def render_local(self, name:str, size:int):
     return self.smem_align + self.smem_prefix + f"float {name}[{size}];"
 
   def render_for(self, expr: str, _min:Union[int,str], _max:Union[int,str]) -> str:
-    return f"for (int {expr} = {_min}; {expr} < {_max}; ++{expr}) {{"
+    return f"for ({self.generic_var_prefix if self.generic_var_prefix else 'int'} {expr} = {_min}; {expr} < {_max}; {f'{expr}++' if self.generic_var_prefix else f'++{expr}'}) {{"
```

Comment:
Huh, why the expr++ ++expr thing?

Reply:
setting default for `generic_var_prefix` affects a alu, define acc, etc, so it will be picked against `dtype.name` in those places:
`kk(f"{lang.generic_var_prefix if lang.generic_var_prefix else dtype.name} {ssa(u,'val')} = {val};")`

Reply:
setting default for generic_var_prefix affects a alu, define acc, etc, so it will be picked against dtype.name in those places:
kk(f"{lang.generic_var_prefix if lang.generic_var_prefix else dtype.name} {ssa(u,'val')} = {val};")

The ++ thing -> wgsl doesn't have pre-increment, but i can just make it post-increment for all.

Reply:
We can shave off 2 more lines if we remove the `render_if` which is only custom in wgsl because of the explicit bool cast. We can make that cast for all `render_if`s, which does not add extra lines.